flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2830,14 +2799,14 @@ struct SelectRankOpConversion : public FIROpConversion<fir::SelectRankOp> {
   }
 };
 
+/// Lower `fir.select_type` to LLVM IR dialect.
 struct SelectTypeOpConversion : public FIROpConversion<fir::SelectTypeOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::SelectTypeOp select, OperandTy operands,
+  matchAndRewrite(fir::SelectTypeOp select, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    mlir::emitError(select.getLoc(),
-                    "fir.select_type should have already been converted");
+    TODO(select.getLoc(), "fir.select_type codegen");
```

Comment:
This one is intended to be a hard error. I don't expect code gen to ever lower this high-level construct.

Reply:
Reverted to the original error